org.eclipse.xtext.common.types.ui/src/org/eclipse/xtext/common/types/xtext/ui/JdtTypesProposalProvider.java

```diff
@@ -400,15 +403,69 @@ public boolean isCanceled() {
 			Iterable<IEObjectDescription> allDirtyTypes = dirtyStateManager.getExportedObjectsByType(TypesPackage.Literals.JVM_TYPE);
 			for(IEObjectDescription description: allDirtyTypes) {
 				QualifiedName qualifiedName = description.getQualifiedName();
-				if (filter.accept(Flags.AccPublic, qualifiedName.skipLast(1).toString().toCharArray(), qualifiedName.getLastSegment().toCharArray(), new char[0][0], description.getEObjectURI().toPlatformString(true))) {
+				final int modifiers = getDirtyStateModifiers(context, description);
+				if (filter.accept(modifiers, qualifiedName.skipLast(1).toString().toCharArray(), qualifiedName.getLastSegment().toCharArray(), new char[0][0], description.getEObjectURI().toPlatformString(true))) {
 					String fqName = description.getQualifiedName().toString();
-					createTypeProposal(fqName, Flags.AccPublic, fqName.indexOf('$') > 0, proposalFactory, myContext, scopeAware, jvmTypeProvider, valueConverter);
+					createTypeProposal(fqName, modifiers, fqName.indexOf('$') > 0, proposalFactory, myContext, scopeAware, jvmTypeProvider, valueConverter);
 				}
 			}
 		}
 	}
 	
 	/**
+	 * Compute the JVM modifiers that corresponds to the given description.
+	 *
+	 * <p>This function fixes the issue related to the missed modifiers given to the content assist.
+	 *
+	 * @param context the current content assist context.
+	 * @param description the description.
+	 * @return the JVM modifiers.
+	 * @since 2.11
+	 */
+	protected int getDirtyStateModifiers(ContentAssistContext context, IEObjectDescription description) {
+		EObject eobject = description.getEObjectOrProxy();
+		if (eobject.eIsProxy()) {
+			eobject = EcoreUtil.resolve(eobject, context.getResource().getResourceSet());
+		}
+		int accessModifiers = Flags.AccPublic;
+		int otherModifiers = 0;
+		if (eobject instanceof JvmMember) {
+			final JvmMember member = (JvmMember) eobject;
+			if (member.isDeprecated()) {
```

Comment:
We should use org.eclipse.xtext.common.types.util.DeprecationUtil.isDeprecatedMember(JvmMember) here. It also checks for @Deprecated annotation